packages/suite/src/middlewares/wallet/storageMiddleware.ts

```diff
@@ -227,9 +229,27 @@ const storageMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => {
                 case METADATA.DISABLE:
                 case METADATA.ADD_PROVIDER:
                 case METADATA.REMOVE_PROVIDER:
-                    api.dispatch(storageActions.saveMetadata());
+                    api.dispatch(storageActions.saveMetadataSettings());
                     break;
-
+                case METADATA.SET_ERROR_FOR_DEVICE: {
+                    const device = selectDeviceByState(api.getState(), action.payload.deviceState);
+                    if (isDeviceRemembered(device) && device) {
```

Comment:
nitpick - if there is no device, it's never remembered...
```suggestion
                    if (isDeviceRemembered(device)) {
```

Reply:
unfortunally, without && device might still be undefined in eyes of typescript on the following line. interestingly, if I use this condition directly (copied from isDeviceRemembered)
```
!!device?.remember
```
&& device is not needed anymore.